geopandas/tools/overlay.py

```diff
@@ -362,7 +362,12 @@ def _make_valid(df):
             # improve performance (we only need to collect geometries in their
             # respective Multi version)
             dissolved = exploded.dissolve(by="level_0")
-            result.loc[is_collection, geom_col] = dissolved[geom_col].values
+            # https://pandas.pydata.org/docs/whatsnew/v1.5.0.html#inplace-operation-when-setting-values-with-loc-and-iloc
+            if is_collection.sum() == len(result):
+                result[geom_col] = dissolved[geom_col].values
+            else:
+                result.loc[is_collection, geom_col] = dissolved[geom_col].values
```

Comment:
I really appreciate the effort to go through and silence the warnings in our CI, this is something I made a small start on a while back and was hoping to progress further, but haven't had the chance.

I understand this stops the warning being emitted, but I don't see why we need to preserve the old behaviour, result is an intermediate dataframe so we shouldn't need to care about the old values?

Might be missing something but wouldn't it make more sense to filter these warnings - as they do not appear to be material in this situation, and we can remove the filter and update to this case being handled for free when pandas makes the switch?

Same question for the case in clip above.

Reply:
I agree this is a very specific edge case.  I am looking at it that it is a *FutureWarning* so at some point a release of **pandas** will start throwing it as an error.  Hence safety / quality is ensure that **geopandas** is not using **pandas** in a way that will break.  When I googled this warning I did not find this case anywhere where `mask` is the complete index,  this warning gets generated.   A possible alternative is to raise it as an issue against **pandas**

Reply:
As I understand it, the warning is about a change in behaviour, it is not a deprecation, and it will not raise an error in future.
"In a future version, setting an entire column with iloc will attempt to operate inplace." - in this case, as far as I can tell we do not care if the operation happens inplace or not, so there is no requirement to preserve the old behaviour.